src/screens/BallotScreen.tsx

```diff
@@ -1,75 +1,282 @@
+  outline: none;
+  cursor: pointer;
+  left: ${({ rect }) => rect.x}px;
+  top: ${({ rect }) => rect.y}px;
+  width: ${({ rect }) => rect.width}px;
+  height: ${({ rect }) => rect.height}px;
+
+  ::after {
+    position: absolute;
+    top: ${({ rect }) => (rect.height - 28) / 2}px;
+    right: ${30 - 28}px;
+    width: 28px;
+    height: 28px;
+    content: '${({ marked, pending }) =>
+      marked ? '☑️' : pending ? '☒' : '☐'}';
+    color: ${({ marked, pending }) =>
+      marked ? '#006600ee' : pending ? '#660000ee' : '#006600ee'};
+  }
+
+  :hover {
+    background-color: ${({ marked }) =>
+      !marked ? 'rgba(71,167,75,.2)' : 'rgba(71,167,75,.4)'};
   }
+`
 
-  if (mark.type === 'yesno') {
-    return mark.option
+const ListWithEmptyState = ({
+  empty,
+  children,
+}: {
+  empty: React.ReactElement
+  children: readonly React.ReactChild[]
+}) => {
+  if (children.length === 0) {
+    return empty
   }
+
+  return <React.Fragment>{children}</React.Fragment>
 }
 
 export default function BallotScreen() {
-  const { batchId, ballotId } = useParams<{
-    batchId?: string
+  const { ballotId } = useParams<{
     ballotId?: string
   }>()
-  const [ballot, setBallot] = useState<Ballot>()
+  const [state, setState] = useState<workflow.State>(workflow.init())
+  const ballot =
+    state.type === 'init' || state.type === 'failed' ? undefined : state.ballot
 
   useEffect(() => {
-    if (!ballot) {
-      fetchJSON<Ballot>(`/scan/batch/${batchId}/ballot/${ballotId}`).then(
-        setBallot
+    if (state.type === 'init') {
+      fetchJSON<ReviewBallot>(`/scan/hmpb/ballot/${ballotId}`).then(
+        (newBallot) => {
+          setState(workflow.fetchedBallotInfo(state, newBallot))
+        }
       )
     }
-  }, [ballot, batchId, ballotId, setBallot])
+  }, [ballotId, state, setState])
+
+  const scale = useCallback(
+    <T extends number | undefined>(value: T): T =>
+      (typeof value === 'number'
+        ? value * (800 / (ballot?.ballot.image.width ?? 800))
+        : value) as T,
+    [ballot]
+  )
+
+  const scaleRect = useCallback(
+    (rect: Rect) => ({
+      x: scale(rect.x),
+      y: scale(rect.y),
+      width: scale(rect.width),
+      height: scale(rect.height),
+    }),
+    [scale]
+  )
+
+  const onContestOptionClick = useCallback<
+    React.MouseEventHandler<HTMLElement>
+  >(
+    (event) => {
+      if (state.type !== 'review') {
+        return
+      }
+
+      const { contestId, contestOptionId } = event.currentTarget.dataset
+      const contest = state.ballot.contests.find(({ id }) => id === contestId)
+      const contestOption = contest?.options.find(
+        ({ id }) => id === contestOptionId
+      )
+
+      if (contest && contestOption) {
+        setState((previous) =>
+          workflow.toggle(previous, contest, contestOption)
+        )
+      }
+    },
+    [state]
+  )
+
+  const onContestOptionKeyPress = useCallback<
+    React.KeyboardEventHandler
+  >(() => {
+    // noop
+  }, [])
```

Comment:
What's this for?

Reply:
Cruft leftover from when I had a raw `<button>` and eslint was telling me I needed a key event handler if I had a tab index.